sdks/python/apache_beam/runners/interactive/pipeline_instrument.py

```diff
@@ -205,7 +351,7 @@ def _process(self, pcoll):
     v = PreprocessVisitor(self)
     self._pipeline.visit(v)
 
-  def _write_cache(self, pcoll):
+  def _write_cache(self, pipeline, pcoll):
```

Comment:
Could we omit this from the signature since all mutation happens to the same runner pipeline instance instantiated by this instrument instance? It should be self contained.

Reply:
The PipelineInstrument now can create two pipelines, the background caching pipeline and the pipeline fragment. This _write_cache needs to be generalized to be able to work on different pipelines (not just self._pipeline), hence the change.

Reply:
Got it! Thanks!